src/allotropy/parsers/msd_workbench/msd_workbench_structure.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+import pandas as pd
+
+from allotropy.allotrope.models.shared.components.plate_reader import SampleRoleType
+from allotropy.allotrope.schema_mappers.adm.plate_reader.rec._2024._06.plate_reader import (
+    Measurement,
+    MeasurementGroup,
+    MeasurementType,
+    Metadata,
+)
+from allotropy.exceptions import AllotropeConversionError
+from allotropy.parsers.constants import DEFAULT_EPOCH_TIMESTAMP, NOT_APPLICABLE
+from allotropy.parsers.msd_workbench.constants import (
+    LUMINESCENCE,
+    LUMINESCENCE_DETECTOR,
+    SAMPLE_ROLE_TYPE_MAPPING,
+    SOFTWARE_NAME,
+)
+from allotropy.parsers.utils.pandas import SeriesData
+from allotropy.parsers.utils.uuids import random_uuid_str
+
+
+@dataclass(frozen=True)
+class Header:
```

Comment:
I think it's cleaner if you just remove this class and pass `file_name` into `create_metadata`